aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/Translator.java

```diff
@@ -197,6 +236,42 @@ public static CreateDbInstanceRequest createDbInstanceRequest(
                 .build();
     }
 
+    public static ModifyDbInstanceRequest modifyDbInstanceRequestV12(
+            final ResourceModel previousModel,
+            final ResourceModel desiredModel,
+            final Boolean isRollback
+    ) {
+        final ModifyDbInstanceRequest.Builder builder = ModifyDbInstanceRequest.builder()
+                .dbInstanceIdentifier(desiredModel.getDBInstanceIdentifier())
+                .dbInstanceClass(desiredModel.getDBInstanceClass())
+                .dbSecurityGroups(desiredModel.getDBSecurityGroups())
+                .applyImmediately(Boolean.TRUE)
+                .masterUserPassword(desiredModel.getMasterUserPassword())
+                .dbParameterGroupName(desiredModel.getDBParameterGroupName())
+                .backupRetentionPeriod(desiredModel.getBackupRetentionPeriod())
+                .preferredBackupWindow(desiredModel.getPreferredBackupWindow())
+                .preferredMaintenanceWindow(desiredModel.getPreferredMaintenanceWindow())
+                .multiAZ(desiredModel.getMultiAZ())
+                .engineVersion(desiredModel.getEngineVersion())
+                .allowMajorVersionUpgrade(desiredModel.getAllowMajorVersionUpgrade())
+                .autoMinorVersionUpgrade(desiredModel.getAutoMinorVersionUpgrade())
+                .optionGroupName(desiredModel.getOptionGroupName());
+
+        if (previousModel != null && BooleanUtils.isTrue(isRollback)) {
+            builder.allocatedStorage(
+                    canUpdateAllocatedStorage(previousModel.getAllocatedStorage(), desiredModel.getAllocatedStorage()) ? getAllocatedStorage(desiredModel) : getAllocatedStorage(previousModel)
+            );
+            builder.iops(
+                    canUpdateIops(previousModel.getIops(), desiredModel.getIops()) ? desiredModel.getIops() : previousModel.getIops()
```

Comment:
Yeah, that one and `canUpdateAllocatedStorage` are tricky. The only job this check has is to ensure that upon a rollback we're not setting allocated storage and iops to a lower value: this would be rejected. Hence we don't care that much about the input validity or anything (upon a rollback both values have been validated by RDS API already), but simply care about non-reducing these digits.